### PR TITLE
refactor(jans-cedarling): split policy store into internal and legacy loading structures

### LINKED ISSUE #13591: refactor(jans-cedarling): split policy store into internal and legacy loading structures
Refactor the policy store: split it into an internal structure and a structure used for loading legacy JSON policy store data.

It allows us to add updates to the current structure without breaking the outdated loading method

### PR DESCRIPTION
**Pull request by olehbozhok**
### Prepare

- [x] Read [PR guidelines](https://github.com/JanssenProject/jans/blob/main/docs/CONTRIBUTING.md#prs)
- [x] Read [license information](https://github.com/JanssenProject/jans/blob/main/LICENSE)

-------------------

#### Target issue

closes #13591

#### Implementation Details

`PolicyStore` was originally used both as an internal representation and as the deserialization target for the legacy Agama Lab JSON/YAML format. This coupling meant any change to the internal representation risked breaking legacy format parsing.

**New `legacy_store` module** (`cedarling/src/common/policy_store/legacy_store/`):
- `LegacyAgamaPolicyStore`, `LegacyPolicyStore` — top-level deserialization structs
- `LegacyCedarSchema`, `LegacyPoliciesContainer`, `LegacyTrustedIssuer`, `LegacyTokenEntityMetadata`, `LegacyDefaultEntitiesWithWarns` — fully independent copies of all types needed for JSON/YAML deserialization
- Own copies of helper enums (`Encoding`, `ContentType`, `PolicyContentType`) and utility traits (`PartitionResult`)
- `From<LegacyX> for X` conversions for each type
- All legacy deserialization tests moved here

**Cleaned up internal types:**
- `PolicyStore` — removed `Deserialize` impl
- `AgamaPolicyStore` — removed entirely (only existed for deserialization)
- `CedarSchema` — removed `Deserialize` impl, helper types (`EncodedSchema`, `MaybeEncoded`), and error enum
- `PoliciesContainer` — removed `Deserialize` impl; replaced `raw_policy_info: HashMap<String, RawPolicy>` with `descriptions: HashMap<String, String>` (raw policy content was unused at runtime)
- `TrustedIssuer` — removed `Deserialize` derive and serde attributes; made `new()` constructor `pub(crate)` (was `#[cfg(test)]` only)
- `DefaultEntitiesWithWarns` — removed `Deserialize` impl
- `Encoding`/`ContentType` enums removed from `common/mod.rs`

**Updated call sites:**
- `init/policy_store.rs` — deserializes into `LegacyAgamaPolicyStore`, converts via `.into()`
- `cedar_schema/mod.rs` tests — use legacy types
- `entity_builder/build_iss_entity.rs` tests — use `TrustedIssuer::new()` instead of `serde_json::from_value`
-------------------
### Test and Document the changes
- [x] Static code analysis has been run locally and issues have been fixed
- [x] Relevant unit and integration tests have been added/updated
- [ ] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [x] **I confirm that there is no impact on the docs due to the code changes in this PR.**

## Summary by CodeRabbit

* **Refactor**
  * Moved legacy policy-store and schema parsing into a dedicated legacy path and simplified how legacy formats are handled.

* **Chores**
  * Removed several direct-deserialize code paths; initialization now consistently uses the legacy-compatible import flow and streamlined parsing surfaces.

* **Tests**
  * Added end-to-end and negative tests for legacy JSON/YAML decoding, base64/UTF-8 edge cases, version parsing, and targeted error messages.

### CODE REVIEW

**Comment by mo-auto** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/janssenproject/pr-checks/4848fbc9-71a2-413d-84b0-27c3a6caa4b2?refs=4fe4983b92b435035090e398ae89ae23c5c317e6&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs [hunk 1]
```diff
@@ -0,0 +1,630 @@
...
+            match r {
+                Ok(v) => ok.push(v),
+                Err(e) => errs.push(e),
+            }
+        }
+        (ok, errs)
+    }
+}
+
+impl<T, E, I> PartitionResult<T, E> for I where I: Iterator<Item = Result<T, E>> {}
```
Comment (dagregi) [inline jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs]: This is duplicated it already exists in `crate::common`

Comment (olehbozhok) [reply jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs]: Thanks, I missed this after reviewing AI
fixed in 961ff2034c08f1c264c15432ab2b7c6880d8109d

File: jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs [hunk 2]
```diff
@@ -0,0 +1,615 @@
...
+
+        let decoded_body = match encoded_schema.encoding {
+            Encoding::None => encoded_schema.body,
+            Encoding::Base64 => {
+                let buf = BASE64_STANDARD.decode(encoded_schema.body).map_err(|err| {
+                    de::Error::custom(format!("{}: {}", ParseCedarSchemaSetMessage::Base64, err))
+                })?;
+                String::from_utf8(buf).map_err(|err| {
+                    de::Error::custom(format!("{}: {}", ParseCedarSchemaSetMessage::Utf8, err))
+                })?
```
[the remark this answers is not in the document]
Comment (olehbozhok) [reply jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs:196]: This code is not a new feature; it is just to have backward compatibility with legacy. So it is fine. Unit tests are passing.

File: jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs [hunk 3]
```diff
@@ -0,0 +1,630 @@
...
+                let (schema_fragment, _warning) =
+                    cedar_policy::SchemaFragment::from_cedarschema_str(&decoded_body).map_err(
+                        |err| {
+                            de::Error::custom(format!(
+                                "{}: {}",
+                                ParseCedarSchemaSetMessage::Parse,
+                                err
+                            ))
+                        },
+                    )?;
```
[the remark this answers is not in the document]
Comment (olehbozhok) [reply jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs:212]: it is OK, this structure is used only for legacy loading

File: jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs [hunk 4]
```diff
@@ -0,0 +1,630 @@
...
+    }
+}
+
+pub(crate) fn parse_cedar_version(value: impl serde::Serialize) -> Result<Version, String> {
+    let Ok(serde_json::Value::String(s)) = serde_json::to_value(&value) else {
+        return Err("cedar_version must be a string".to_string());
+    };
+    let s = s.strip_prefix('v').unwrap_or(&s);
+    Version::parse(s).map_err(|e| format!("error parsing cedar version :{e}"))
+}
```
[the remark this answers is not in the document]
Comment (olehbozhok) [reply jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs]: function parse_cedar_version  was removed

[the remark this answers is not in the document]
Comment (olehbozhok) [reply jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs]: We don't have it anymore

File: jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs [hunk 5]
```diff
@@ -0,0 +1,615 @@
...
+            .policies()
+            .map(|p| (p.id().clone(), p))
+            .collect();
+        let other_policies: BTreeMap<_, _> = other
+            .policy_set
+            .policies()
+            .map(|p| (p.id().clone(), p))
+            .collect();
+        self_policies == other_policies
+    }
```
[the remark this answers is not in the document]
Comment (olehbozhok) [reply jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs:342]: fixed

File: jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs [hunk 6]
```diff
@@ -0,0 +1,615 @@
...
+            description: v.description,
+            cedar_version: v.cedar_version,
+            schema: v.schema.into(),
+            policies: v.policies.into(),
+            trusted_issuers: v
+                .trusted_issuers
+                .map(|issuers| issuers.into_iter().map(|(k, v)| (k, v.into())).collect()),
+            default_entities: v.default_entities.into(),
+        }
+    }
```
[the remark this answers is not in the document]
Comment (olehbozhok) [reply jans-cedarling/cedarling/src/common/policy_store/legacy_store/mod.rs:557]: Since there's no root-level cedar_version to lose or propagate, this bug doesn't exist in the current code. No fix is needed.

**Review by haileyesus2433 — DISMISSED** [review]

**Review by dagregi — DISMISSED** [review]

**Review by haileyesus2433 — APPROVED** [review]
